src/main/java/com/github/rnewson/couchdb/lucene/util/Analyzers.java

```diff
@@ -186,13 +280,313 @@ protected TokenStreamComponents wrapComponents(String fieldName, TokenStreamComp
         }
     }
 
+	public static Analyzer fromSpec(final JSONObject json, final String analyzerKey) throws JSONException {
+		JSONObject spec = json.optJSONObject(analyzerKey);
+		if (spec != null) {
+			return getAnalyzer(spec);
+		} else {
+			return getAnalyzer(json.optString(analyzerKey, Constants.DEFAULT_ANALYZER));
+		}
+	}
+
+	public static Analyzer fromSpec(final JSONObject json) throws JSONException {
+		return fromSpec(json, Constants.ANALYZER);
+	}
+
+	/*
+	 * called from DatabaseIndexer when handling an http search request
+	 */
+	public static Analyzer fromSpec(String str) throws JSONException {
+		if (str == null) {
+			return getAnalyzer(Constants.DEFAULT_ANALYZER);
+		} 
+		
+		str = str.trim();
+		
+		if (str.startsWith("{")) {
+			try {
+				return getAnalyzer( new JSONObject(str) );
+			
+			} catch (Exception ex) {
+				return getAnalyzer( Constants.DEFAULT_ANALYZER );
+			}
+		}
+		
+		return getAnalyzer(str);
+	}
+
     public static Analyzer getAnalyzer(final String str) throws JSONException {
         final String[] parts = str.split(":", 2);
         final String name = parts[0].toUpperCase();
         final String args = parts.length == 2 ? parts[1] : null;
         return Analyzers.valueOf(name).newAnalyzer(args);
     }
 
+	public static Analyzer getAnalyzer(final JSONObject json) {
+		String className = json.optString(Constants.CLASS);
+		JSONArray params = json.optJSONArray(Constants.PARAMS);
+
+		Analyzer newAnalyzer = null;
+
+		if (className == null || className.isEmpty()) {
+			Iterator<?> it = json.keys();
+			if (it.hasNext()) {
+				String key = (String) it.next();
+				String args = json.optString(key);
+				
+				JSONObject obj = json.optJSONObject(key);
+				try {
+					if (obj != null) {
+						return Analyzers.valueOf(key.toUpperCase()).newAnalyzer(obj);
+					} else {
+						return Analyzers.valueOf(key.toUpperCase()).newAnalyzer(args);
+					}
+				} catch (Exception ex) { }
+			}
+
+			logger.error("Lucene index: analyzer class name is not defined");
+			return null;
+		}
+
+		// is the class accessible?
+		Class<?> clazz = null;
+		try {
+			clazz = Class.forName(className);
+		} catch (ClassNotFoundException e) {
+			logger.error(String.format("Lucene index: analyzer class %s not found. (%s)", className, e.getMessage()));
+			return null;
+		}
+
+		// Is the class an Analyzer?
+		if (!Analyzer.class.isAssignableFrom(clazz)) {
+			logger.error(String.format("Lucene index: analyzer class has to be a subclass of %s", Analyzer.class.getName()));
```

Comment:
as I look at all these cases and their handling, I think a better approach would be to throw an exception, with a useful message, that we catch and log at the point where we call getAnalyzer(obj) above.